### PR TITLE
zarr IO for templates object

### PR DESCRIPTION
**Pull request by h-mayorquin**
Related to the hybrid project as requested.

### CODE REVIEW

**Comment by yger** [comment]
Thanks a lot! Just because I'm also playing with the Templates, with such a PR, will be able to load/save templates as we can currently do with sorting/recordings ? Can we have a save() function that will keep the same syntax as for other objects?

File: src/spikeinterface/core/template.py [hunk 1]
```diff
@@ -165,6 +165,67 @@ def from_dict(cls, data):
...
+
+        if self.sparsity_mask is not None:
+            zarr_group.create_dataset("sparsity_mask", data=self.sparsity_mask)
+
+        if self.probe is not None:
+            probe_dict = self.probe.to_dict(array_as_list=True)
+            zarr_group.attrs["probe"] = probe_dict
+
+    @classmethod
+    def load_from_open_zarr_storage(cls, base_root):
```
Comment (alejoe91) [inline src/spikeinterface/core/template.py:190]:
```suggestion
    def load_from_zarr_group(cls, base_root):
```
A zarr open storage is a zarr group, no?

**Comment by alejoe91** [comment]
@yger for now let's keep the `to_zarr`/`from_zarr`. We can refactor the calls later

File: src/spikeinterface/core/template.py (hunk 1 above)
Comment (alejoe91) [reply src/spikeinterface/core/template.py:190]: It doesn't need to be root. For example, the new sorting result could save this in the `results/templates` group!

Comment (h-mayorquin) [reply src/spikeinterface/core/template.py:190]: OK, let's go for the suggested naming and say that the structure of the templates as defined by the writing function should anchored in that group.

Comment (h-mayorquin) [reply src/spikeinterface/core/template.py]: Done.

**Comment by alejoe91** [comment]
@samuelgarcia ready to merge on my side. Wanna take a look?

**Comment by h-mayorquin** [comment]
Related to #248 and #250

**Comment by samuelgarcia** [comment]
Thanks a lot for this this will be usefull.
Amybe we could add also the possibility to give an existing zarr folder and append this temapltes to specific sub group.
We have this in zarr sorting:
https://github.com/SpikeInterface/spikeinterface/blob/main/src/spikeinterface/core/zarrextractors.py#L176
https://github.com/SpikeInterface/spikeinterface/blob/main/src/spikeinterface/core/zarrextractors.py#L324

This can be convinient to save templates + other suff.

**Comment by h-mayorquin** [comment]
@samuelgarcia I added the requested method. 

This should be ready save using the zarr machinery developed on the probeinterface side:
https://github.com/SpikeInterface/probeinterface/pull/250

**Review by alejoe91 — APPROVED** [review]

**Pull request merged**